solutions/usaco-696.mdx

```diff
@@ -2,10 +2,102 @@
 id: usaco-696
 source: USACO Platinum 2017 January
 title: Promotion Counting
-author: Benjamin Qi
+author: Benjamin Qi, Timothy Gao
 ---
 
-## Merging Indexed Sets
+## Solution 1: Euler Tour
+**Time Complexity:** $\mathcal{O}(N\log N)$
+
+Recall from the euler tour module that each subtree is represented as a contiguous interval in it's euler tour. With this knowledge, we can compress the tree into an array. Now, the problem becomes for each node $i$, look for how many elements are less than $tour[i]$ from $i$ to $tout[i]$, where $tout[i]$ is the end of the subtree interval for node $i$. By processing the nodes in decreasing order, this is a problem we can solve in $\mathcal{O}(N\log N)$ time with a binary-indexed tree.
+
+```java
+import java.io.*;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public class promote {
+    
+    public static void main(String[] args) throws IOException {
```

Comment:
use tabs instead of spaces

Reply:
why was this resolved? this was not addressed.

Reply:
Oops. I thought I had changed this. Curious though: Why is this necessary?

Reply:
it's just a code convention

https://usaco.guide/general/adding-solution?lang=cpp#general

Reply:
i see